### ISSUE TITLE
Add privacy auditing to training example

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#48]
## Summary
Add optional privacy auditing to `train_causal_lm.py` using the `opaque.auditing` module. Auditing empirically estimates the privacy leakage by comparing model behavior on member vs non-member data.

## Current State
- `opaque.auditing` module has `setup()` and `evaluate()` API (108 tests, well-tested)
- Uses `score_by_loss()` with `torch.func.vmap` for per-example loss computation
- No example or training script uses auditing

## Implementation
1. `--audit` flag enables post-training auditing (default: disabled)
2. `--audit-canaries N` sets number of canary examples (default: 100)
3. After training, run auditing:
   ```python
   audit_result = auditing.evaluate(model, train_set, test_set, ...)
   print(f"Empirical ε: {audit_result.epsilon_lower_bound:.2f}")
   print(f"Theoretical ε: {acct.epsilon_at(delta):.2f}")
   ```
4. Compare empirical vs theoretical epsilon (empirical should be ≤ theoretical)

## What NOT to Do
- Do NOT run auditing during training (too expensive)
- Do NOT log canary membership to W&B (privacy risk)

## Acceptance Criteria
- [ ] `--audit` flag runs post-training auditing
- [ ] Empirical epsilon reported alongside theoretical
- [ ] Graceful handling if auditing fails (warn, don't crash)
- [ ] Works with LoRA models

## Depends on
- #47 (accounting must be in place first)

## Parent
- #65
